### PR TITLE
feat: warn response format

### PR DESCRIPTION
**Pull request by loks0n**
Adds support for new header `X-Appwrite-Warning: Invalid SDK format version requested; Wrong content-type requested`

Contents can be seperated by `;` and is logged in a warning message to the console

### CODE REVIEW

File: templates/php/src/Client.php.twig [hunk 1]
```diff
@@ -176,6 +176,13 @@ class Client
         $responseBody   = curl_exec($ch);
         $contentType    = $responseHeaders['content-type'] ?? '';
         $responseStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        $warnings = $responseHeaders['x-{{spec.title | caseLower}}-warning'] ?? '';
+        if ($warnings) {
+            foreach(explode(';', $warnings) as $warning) {
+                trigger_error($warning, E_USER_WARNING);
```
Comment (lohanidamodar) [inline templates/php/src/Client.php.twig:183]: I don't think `trigger_error` is right, I think it throws exception from what I understood here https://www.php.net/manual/en/function.trigger-error.php

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
shared some concerns.

File: templates/php/src/Client.php.twig (hunk 1 above)
Comment (loks0n) [reply templates/php/src/Client.php.twig:183]: Screenshot 2024-05-21 at 10 19 41
Seems to behave how we need based on my playground testing

File: templates/dart/lib/src/client_mixin.dart.twig [hunk 2]
```diff
@@ -67,6 +67,12 @@ class ClientMixin {
 
   Response prepareResponse(http.Response res, {ResponseType? responseType}) {
     responseType ??= ResponseType.json;
+
+    String? warnings = res.headers['x-{{ spec.title | lower }}-warning'];
+    if (warnings != null) {
+      warnings.split(';').forEach((warning) => print(warning));
```
[the remark this answers is not in the document]
Comment (loks0n) [reply templates/dart/lib/src/client_mixin.dart.twig]: Done!

**Review by lohanidamodar — APPROVED** [review]

File: templates/php/src/Client.php.twig (hunk 1 above)
Comment (loks0n) [reply templates/php/src/Client.php.twig]: Done!

**Review by abnegate — COMMENTED** [review]
Some languages are missing the `Warning: ` prefix

File: templates/apple/Sources/Client.swift.twig [hunk 3]
```diff
@@ -278,6 +278,14 @@ open class Client {
             timeout: .seconds(30)
         )
 
+        let warnings = response.headers["x-{{ spec.title | lower }}-warning"].first
+        if warnings != nil {
```
[the remark this answers is not in the document]
Comment (loks0n) [reply templates/apple/Sources/Client.swift.twig]: done

File: templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig [hunk 4]
```diff
@@ -523,6 +523,14 @@ class Client @JvmOverloads constructor(
                     it.cancel(error)
                     return
                 }
+
+                val warnings = response.headers["x-{{ spec.title | lower }}-warning"]
+                if (warnings != null) {
+                    warnings.split(";").forEach { warning ->
+                        System.out.println("Warning: $warning")
```
[the remark this answers is not in the document]
Comment (loks0n) [reply templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig]: done

**Review by abnegate — APPROVED** [review]

**Pull request merged**